### DISCUSSION TITLE
Is there a way to prevent sending automatic announcements?

### DISCUSSION THREAD

**Discussion by ThomasFreedman** [markqvist/Reticulum#293] [Help & Questions]
I'm noticing that at some point in the process of initializing RNS and creating a destination, that RNS sends out an announcement. I'm curious if that behavior can be eliminated or what the ramifications of doing so would be.

The startup announcement is undesireable, as I want to only send announcements on demand. I don't see any other way widely to broadcast an encrypted message network wide without a known destination besides using the RNS announcement method.

I suppose it's not really a big deal, to have the intended receivers ignore the initial announcement, since it won't include the data I'll be sending announcements for on demand.

**Comment by faragher** [comment]
Short version: in a properly functioning system with a known destination address, direct delivery is possible. I don't believe routed delivery is possible.

Long version: An announce does three things. 

First, it announces the node's presence. This really doesn't matter if you have the address known ahead of time. 

Second, it propagates the cryptographic information needed to send a message to the recipient. You can't send a message without this key. However, the sender can request a path and key from the network, meaning an announce isn't required to be seen by the sender prior to sending a message. That is the function of request_path in the link example. [Example here.](https://github.com/markqvist/Reticulum/blob/master/Examples/Link.py) 

Third, it establishes a route. An announce is recorded by nodes that pick it up, and propagation nodes spread that information. If Alice announces and Bob sees it, then Bob will announce. Alice ignores the announce, but Charlie, who sees Bob but not Alice, stores that the address is available through Bob, then announces that it knows a path to Alice. Until every node (propagation node?) has heard the announce or the limit is reached, this continues.  So requesting a route is only possible if the route exists, but as in the link example, a direct connection can query the local net or propagation node to get the data on demand.

So yes, you can eliminate announcements (although I've never checked to see if auto-announce is a thing, since the examples I've used all announce manually) and have reasonable functionality. If you announce at some point to register the path with the network, then you'd have full functionality. 

There's some information regarding the intended uses of interface modes on page 56 of the documentation if you're building devices that aren't meant to be full-fledged nodes. I'd look into the access_point mode, as it's designed to remain quiet and not auto-announce. It may not exactly suit your needs, as an AP does handle path requests.

Also, for reference, paths do go stale, meaning a reannounce is required. (Per transport.py)

Default: One week
Access Point: One day
Roaming: Six hours

Bear in mind, I'm not Mark, any of this could be wrong. This is simply my experience.

**Comment by markqvist** [comment]
> I'm noticing that at some point in the process of initializing RNS and creating a destination, that RNS sends out an announcement.

What you are seeing here is not really an announce being *sent*, per se. You are just seeing local instances connected to a shared instance communicating which destinations are available where to each other. Unless an application actually sends an announce for one of its destinations using the `Destination.announce()` API call, nothing will be sent on the network.

The default behaviour is for everything in Reticulum to be completely silent until asked to send something. But as @faragher says, until an announce for a destination has been sent, only the immediately local network will be able to reach it (unless recursive path resolution is operational on the network segments between the initiator and the unannounced destination. In that case, a path can be recursively resolved by anyone requesting it).

If you really want to compare it to TCP/IP, think of a Destination as a combination of an IP-address and a port. The destination is not universal on a device/system-level like an IP-address is: One system/device/application can have as many (or as few) destinations as required.

**Comment by ThomasFreedman** [comment]
Thanks guys for that info. Good to know what appeared to be an announce (since the announce handler printed one was received) is just seen by local shared instances and not broadcast to RNS globally. I will verify this to be sure, but as I said even if it was an actual announce I'll code the handlers to ignore any that don't have the intended app payload so no worries.
